tools/json_benchmark.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+import multiprocessing
+import time
+import redis
+import sys
+import argparse
+from urllib.parse import urlparse
+import os
+from collections import defaultdict
+import math
+
+'''
+Run JSON benchmark for 3 commands:
+    JSON.SET
+    JSON.GET
+    JSON.TYPE
+We want to the overall time it takes
+to save and access keys that contains
+JSON values with this benchmark.
+This also verify that the basic functionalities
+for using JSON types work correctly
+'''
+
+def ping(r):
+    r.ping()
+
+def jsonset(r, i):
+    key = "json-{}".format(i)
+    r.execute_command('JSON.SET', key, '.', '{"a":123456, "b": "hello", "nested": {"abc": "ffffff", "bfb": null}}')
+
+
+def jsonget(r, i):
+    key = "json-{}".format(i)
+    r.execute_command('JSON.GET', key, '$.a', '$..abc')
+
+def jsontype(r, i):
+    key = "json-{}".format(i)
+    r.execute_command('JSON.TYPE', key, '$.a')
+
+def runWorker(ctx):
+    wpid = os.getpid()
+    print( '{} '.format(wpid))
+
+    rep = defaultdict(int)
+    r = redis.StrictRedis(host=ctx['host'], port=ctx['port'])
+    work = ctx['work']
+    if ctx['pipeline'] == 0:
+        total_count = int(ctx['count'])
+        for i in range(0, total_count):
+            s0 = time.time()
+            jsonset(r, i)
+            s1 = time.time() - s0
+            bin = int(math.floor(s1 * 1000)) + 1
+            rep[bin] += 1
+        for i in range(0, total_count):
+            s0 = time.time()
+            jsonget(r, i)
+            s1 = time.time() - s0
+            bin = int(math.floor(s1 * 1000)) + 1
+            rep[bin] += 1
+        for i in range(0, total_count):
+            s0 = time.time()
+            jsontype(r, i)
+            s1 = time.time() - s0
+            bin = int(math.floor(s1 * 1000)) + 1
+            rep[bin] += 1
+    else:
+        for i in range(0, ctx['count'], ctx['pipeline']):
+            p = r.pipeline()
+            s0 = time.time()
+            for j in range(0, ctx['pipeline']):
+                work(p)
+            p.execute()
+            s1 = time.time() - s0
+            bin = int(math.floor(s1 * 1000)) + 1
+            rep[bin] += ctx['pipeline']
+
+    return rep
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='ReJSON Benchmark', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-c', '--count', type=int, default=100000, help='total number of operations')
+    parser.add_argument('-p', '--pipeline', type=int, default=0, help='pipeline size')
+    parser.add_argument('-w', '--workers', type=int, default=8, help='number of worker processes')
+    parser.add_argument('-u', '--uri', type=str, default='redis://localhost:6379', help='Redis server URI')
+    args = parser.parse_args()
+    uri = urlparse(args.uri)
+
+    r = redis.Redis(host=uri.hostname, port=uri.port)
+
+    pool = multiprocessing.Pool(args.workers)
+    s0 = time.time()
```

Comment:
Interesting to see you went for multiprocessing

It looks like asyncio didn't provide enough throughput? I usually used MSET to make sure it does, but I see json doesn't have it

Reply:
I tried this with async and it was way slower, I was never able to have CPU utilization of more than 10% I think that the async io is not really good choice when trying to max throughput. 
Also I notice that [aio-redis-py](https://github.com/aio-libs/aioredis-py) is not in active development and its not supporting json at all. On the other side when working with json API for python it has an overhead as well.